### ISSUE TITLE
Provider is always marked to be replaced

### ISSUE THREAD

**Issue by arnediekmann** [pulumi/pulumi-eks#207]
Since a week or so whenever I try to run `pulumi up` for a EKS cluster created via 

```typescript
const kubernetesCluster = new eks.Cluster(stackName, {
  instanceType: "t2.medium",
  desiredCapacity: 2,
  minSize: 2,
  maxSize: 4,
  nodeRootVolumeSize: 60
});
```

the preview shows that the provider is set to be replaced, even though the details show no apparent differences:

```
    ++pulumi:providers:kubernetes: (create-replacement)
        [id=cfffaf82-6e8b-4c86-bfa0-c8d60aa2fecf]
        [urn=urn:pulumi:dev::mystack::eks:index:Cluster$pulumi:providers:kubernetes::mystack-dev-eks-k8s]
        kubeconfig: "{\"apiVersion\":\"v1\",\"clusters\":[{\"cluster\":{\"server\":\"<url>\",\"certificate-authority-data\":\"<ca>\"},\"name\":\"kubernetes\"}],\"contexts\":[{\"context\":{\"cluster\":\"kubernetes\",\"user\":\"aws\"},\"name\":\"aws\"}],\"current-context\":\"aws\",\"kind\":\"Config\",\"users\":[{\"name\":\"aws\",\"user\":{\"exec\":{\"apiVersion\":\"client.authentication.k8s.io/v1alpha1\",\"command\":\"aws-iam-authenticator\",\"args\":[\"token\",\"-i\",\"mystack-dev-eksCluster-f0ab8a2\"]}}}]}"
    +-pulumi:providers:kubernetes: (replace)
        [id=cfffaf82-6e8b-4c86-bfa0-c8d60aa2fecf]
        [urn=urn:pulumi:dev::mystack::eks:index:Cluster$pulumi:providers:kubernetes::mystack-dev-eks-k8s]
        kubeconfig: "{\"apiVersion\":\"v1\",\"clusters\":[{\"cluster\":{\"server\":\"<url>\",\"certificate-authority-data\":\"<ca>\"},\"name\":\"kubernetes\"}],\"contexts\":[{\"context\":{\"cluster\":\"kubernetes\",\"user\":\"aws\"},\"name\":\"aws\"}],\"current-context\":\"aws\",\"kind\":\"Config\",\"users\":[{\"name\":\"aws\",\"user\":{\"exec\":{\"apiVersion\":\"client.authentication.k8s.io/v1alpha1\",\"command\":\"aws-iam-authenticator\",\"args\":[\"token\",\"-i\",\"mystack-dev-eksCluster-f0ab8a2\"]}}}]}"
```

As a result Pulumi wants to replace every resource in the EKS cluster, which is not something I want to risk since the resources encompass lots of special resources like CRDs. 

Is there any way to get more details on why Pulumi might consider the provider to have changed?

**Comment by arnediekmann** [comment]
There respective version of Pulumi packages are:

- `@pulumi/aws`: **`0.18.23`**
- `@pulumi/eks`: **`0.18.9`**
- `@pulumi/kubernetes`: **`0.25.2`**
- `@pulumi/pulumi`: **`0.17.25`**

And the CLI I use is version **`0.17.25`**: 

```bash
$ pulumi version
v0.17.25
```

**Comment by lukehoban** [comment]
This looks like the same issue as https://github.com/pulumi/pulumi-kubernetes/issues/645, which was fixed in https://github.com/pulumi/pulumi/pull/2973.  We will be releasing a new version of the `pulumi` CLI today with the fix for this.

**Comment by lukehoban** [comment]
Note that you can workaround this manually by doing the following:
1. `pulumi stack export > stack.json`
2. Find your `pulumi:providers:kubernetes` entry in the statefile.
3. It will most likely not have  any `outputs`, but should have `inputs`
4. Copy it's `inputs` over into it's `outputs` as well and save
5. `pulumi stack import < stack.json`

The CLI fix will automate doing this migration for you if needed.

**Comment by arnediekmann** [comment]
I updated the CLI to version `0.17.26`, but it still marked all resources for replacement. I then tried your workaround and indeed the output of the provider was not set. After copying the input to the output and importing the stack it still wanted to replace the provider this time showing a diff with `=> output<string>`. I.e. it seems that the EKS cluster resource itself is changing though it doesn't really show in the details that it did and why.

I then just tried executing the update hoping that the output would in fact not change and thus omitting the replacements. However the state is now pretty much broken since Pulumi started tinkering with the underlying node launch configuration and is now stuck unable to delete the old one "because it is attached to AutoScalingGroup ...". I think I might actually wipe the cluster and try rebuilding it from scratch.
